### PR TITLE
Faster loading of kernels for execution

### PR DESCRIPTION
**Pull request by DonJayamanne**

### CODE REVIEW

File: src/notebooks/controllers/controllerLoader.ts [hunk 1]
```diff
@@ -120,10 +120,6 @@ export class ControllerLoader implements IControllerLoader, IExtensionSyncActiva
                         this.registration
                     );
                 }
-
-                // Wait for all interpreters to be loaded,
-                // This reduces flickering (changing controllers from one to another).
-                await this.interpreters.waitForAllInterpretersToLoad();
```
Comment (DonJayamanne) [inline src/notebooks/controllers/controllerLoader.ts:126]: This was the problem, we were waiting for all python envs to get discovered before creating the controllers

File: src/kernels/raw/finder/localPythonAndRelatedNonPythonKernelSpecFinder.wrapper.node.ts [hunk 2]
```diff
@@ -50,8 +50,10 @@ export class LocalPythonAndRelatedNonPythonKernelSpecFinderWrapper
     onDidChangeStatus = this._onDidChangeStatus.event;
     private readonly _onDidChangeKernels = new EventEmitter<void>();
     onDidChangeKernels = this._onDidChangeKernels.event;
-    refresh(): Promise<void> {
-        throw new Error('Method not implemented.');
```
Comment (DonJayamanne) [inline src/kernels/raw/finder/localPythonAndRelatedNonPythonKernelSpecFinder.wrapper.node.ts:54]: oops, missed this

File: src/platform/api/pythonApi.ts [hunk 3]
```diff
@@ -435,9 +435,6 @@ export class InterpreterService implements IInterpreterService {
             this.disposables
         );
     }
-    public async waitForAllInterpretersToLoad(): Promise<void> {
```
Comment (DonJayamanne) [inline src/platform/api/pythonApi.ts:438]: removed this method and ensure we call refresh, thats the same
this way we have one less method

File: pythonFiles/vscode_datascience_helpers/kernel_interrupt_daemon.py [hunk 4]
```diff
@@ -336,9 +336,9 @@ def main():
             elif line.startswith("INTERRUPT:"):
                 interrupter.interrupt(int(line.split(":")[2]))
                 print(f"INTERRUPT:{int(line.split(':')[1])}")
-            elif line.startswith("KILL_INTERRUPT:"):
+            elif line.startswith("DISPOSE_INTERRUPT_HANDLE:"):
                 interrupter.close_interrupt_handle(int(line.split(":")[2]))
-                print(f"KILL_INTERRUPT:{int(line.split(':')[1])}")
+                print(f"DISPOSE_INTERRUPT_HANDLE:{int(line.split(':')[1])}")
```
Comment (DonJayamanne) [inline pythonFiles/vscode_datascience_helpers/kernel_interrupt_daemon.py:341]: name was incorrect

File: src/kernels/raw/finder/localPythonAndRelatedNonPythonKernelSpecFinder.node.ts [hunk 5]
```diff
@@ -70,12 +70,12 @@ export class LocalPythonAndRelatedNonPythonKernelSpecFinder extends LocalKernelS
         @inject(IPythonExtensionChecker) extensionChecker: IPythonExtensionChecker,
         @inject(LocalKnownPathKernelSpecFinder)
         private readonly kernelSpecsFromKnownLocations: LocalKnownPathKernelSpecFinder,
-        @inject(IMemento) @named(GLOBAL_MEMENTO) globalState: Memento,
+        @inject(IMemento) @named(WORKSPACE_MEMENTO) memento: Memento,
```
Comment (DonJayamanne) [inline src/kernels/raw/finder/localPythonAndRelatedNonPythonKernelSpecFinder.node.ts:73]: ensure we store stuff in workspace memento, not global
as interpreters and the like are specific to a workspace

**Review by isidorn — APPROVED** [review]

**Pull request merged**